pkg/types/aws_resource_descriptor.go

```diff
@@ -65,4 +68,7 @@ type AWSResourceDescriptor interface {
 	// the resource. This will allow the Kubernetes API server to delete the
 	// underlying CR.
 	MarkUnmanaged(AWSResource)
+	// MarkAdopted places descriptors on the custom resource that indicate the
+	// resource was not created from within ACK.
+	MarkAdopted(AWSResource)
```

Comment:
This should be fine in the resource descriptor interface. Not sure whether in the future the process of marking adoption will mean setting AWS tags on the resource. If it does, then this may need to move to the AWSResourceManager interface.... we'll just have to wait and see.